http.kafka.cqrs/src/test/java/io/aklivity/example/cqrs/ProcessorTest.java

```diff
@@ -0,0 +1,99 @@
+public class ProcessorTest
+{
+    private static final String COMMAND_TOPIC = "task-command";
+    private static final String SNAPSHOT_TOPIC = "task-snapshot";
+    private static final String COMMAND_RESPONSE_TOPIC = "command-response";
+
+    private TopologyTestDriver testDriver;
+
+    private TestInputTopic<String, Command> commandsInTopic;
+    private TestOutputTopic<String, Task> snapshotsOutTopic;
+    private TestOutputTopic<String, Object> commandsResponseTopic;
+
+    @BeforeEach
+    public void setUp()
+    {
+        final StreamsBuilder builder = new StreamsBuilder();
+        final CqrsTopology processor = new CqrsTopology();
+        processor.commandTopic = COMMAND_TOPIC;
+        processor.snapshotTopic = SNAPSHOT_TOPIC;
+        processor.commandResponseTopic = COMMAND_RESPONSE_TOPIC;
+        processor.buildPipeline(builder);
+        final org.apache.kafka.streams.Topology topology = builder.build();
+
+        final Properties props = new Properties();
+        props.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
+        props.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, JsonSerde.class.getName());
+        testDriver = new TopologyTestDriver(topology, props);
+
+        commandsInTopic = testDriver.createInputTopic(COMMAND_TOPIC,
+                new StringSerializer(), new JsonSerializer<>());
+        StringDeserializer keyDeserializer = new StringDeserializer();
+        KafkaJsonDeserializer<Task> snapshotDeserializer = new KafkaJsonDeserializer<>();
+        snapshotDeserializer.configure(Collections.emptyMap(), false);
+        snapshotsOutTopic = testDriver.createOutputTopic(SNAPSHOT_TOPIC,
+                keyDeserializer, snapshotDeserializer);
+        KafkaJsonDeserializer<Object> responseDeserializer = new KafkaJsonDeserializer<>();
+        responseDeserializer.configure(Collections.emptyMap(), false);
+        commandsResponseTopic = testDriver.createOutputTopic(COMMAND_RESPONSE_TOPIC,
+                keyDeserializer, responseDeserializer);
+    }
+
+    @AfterEach
+    public void tearDown()
+    {
+        testDriver.close();
+    }
+
+    @Test
+    public void shouldProcessTaskCommand()
+    {
+        final Headers headers = new RecordHeaders(
+                new Header[]{
+                    new RecordHeader("zilla:correlation-id", "1".getBytes()),
+                    new RecordHeader("idempotency-key", "task1".getBytes()) //TODO: change the key to lower case
+                });
+        commandsInTopic.pipeInput(new TestRecord<>("task1", CreateTaskCommand.builder()
+                .id("task1")
+                .description("Test")
+                .build(), headers));
+        List<KeyValue<String, Object>> response = commandsResponseTopic.readKeyValuesToList();
+        assertEquals(1, response.size());
+        List<KeyValue<String, Task>> snapshots = snapshotsOutTopic.readKeyValuesToList();
+        assertEquals(1, snapshots.size());
+    }
```

Comment:
Let's enhance these tests to cover each of the commands, `POST`, `PUT`, `DELETE`, with and without `if-match`.
Also some negative test cases to verify that command is rejected as expected when `if-match` doesn't align.

Reply:
Sounds good.